internal/providersdkv2/resource_hvn_route.go

```diff
@@ -64,6 +65,35 @@ func resourceHvnRoute() *schema.Resource {
 				Required:    true,
 				ForceNew:    true,
 			},
+			// Optional inputs
+			"azure_config": {
+				Description: "The azure configuration for routing.",
+				Type:        schema.TypeList,
+				ForceNew:    true,
+				MaxItems:    1,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"next_hop_type": {
+							Description: "The type of Azure hop the packet should be sent to. Valid options for Next Hop Type - `VIRTUAL_APPLIANCE` or `VIRTUAL_NETWORK_GATEWAY`",
+							Type:        schema.TypeString,
+							Required:    true,
+							ForceNew:    true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"VIRTUAL_APPLIANCE",
+								"VIRTUAL_NETWORK_GATEWAY",
+							}, true),
+						},
+						"next_hop_ip_address": {
+							Description: "Contains the IP address packets should be forwarded to. Next hop values are only allowed in routes where the next hop type is VIRTUAL_APPLIANCE.",
```

Comment:
Can we use `ValidateDiagFunc` to validate that next hop is only allowed with `VIRTUAL_APPLIANCE`. That will let it be a schema violation not a runtime error

Reply:
👍🏻 Excellent, will do

Reply:
Discussed over slack.
Doesn't look like we can do this with list types at this time.